tools/decdk/README.md

```diff
@@ -0,0 +1,256 @@
+
+```ts
+new lambda.Function(this, 'MyHandler', {
+  handler: 'index.handler',
+  runtime: lambda.Runtime.NodeJS810,
+  code: lambda.Code.asset('./src')
+});
+```
+
+And here's the deCDK version:
+
+```yaml
+MyHandler:
+  Type: "@aws-cdk/aws-lambda.Function"
+  Properties:
+    handler: index.handler
+    runtime: NodeJS810
+    code:
+      asset:
+        path: "./src"
+```
+
+### Polymorphism
+
+Due to the decoupled nature of AWS, The AWS Construct Library highly utilizes polymorphism to expose rich APIs to users. In many cases, APIs would accept an interface of some kind, and various AWS services provide an implementation for that interface. deCDK is able to find all concrete implementation of an interface or an abstract class and offer the user an enum-like experience. The following example shows how this approach can be used to define AWS Lambda events:
+
+```yaml
+MyHandler:
+  Type: "@aws-cdk/aws-lambda.Function"
+  Properties:
+    handler: index.handler
+    runtime: NodeJS810
+    code:
+      asset:
+        path: "./src"
+    events:
+    - DynamoEventSource:
+        table:
+          Ref: Table
+        startingPosition: TrimHorizon
+    - ApiEventSource:
+        method: GET
+        path: "/hello"
+    - ApiEventSource:
+        method: POST
+        path: "/hello"
+    - SnsEventSource:
+        topic:
+          Ref: MyTopic
+```
+
+The keys in the “events” array (“DynamoEventSource”, “ApiEventSource”, “SnsEventSource”) are all names of classes in the AWS Construct Library. The declaration is “Array<IEventSource>”. When deCDK deconstructs the objects in this array, it will create objects of these types and pass them in as IEventSource objects.
+
+### `Fn::GetAtt`
+
+deCDK also supports referencing specific attributes of CDK resources by the intrinsic “Fn::GetAt”. When processing the template, if an Fn::GetAtt is found, and references a CDK construct, the attribute name is treated as a property name of the construct and it's value is used.
+
+The following example shows how to output the “url” property of a @aws-cdk/aws-lambda.Function from above:
+
+```yaml
+Outputs:
+  HelloWorldApi:
+    Description: API Gateway endpoint URL for Prod stage for Hello World function
+    Value:
+      Fn::GetAtt:
+      - MyHandler
+      - url
+```
+
+### Raw CloudFormation
+
+If deCDK doesn't identify a resource type as a CDK resource, it will just pass it through to the resulting output (through a special construct we have in the CDK called `cdk.Include`). This means that any existing CloudFormation/SAM resources (such as `AWS::SQS::Queue`) can be used as-is.
```

Comment:
Don't think implementation details matter for the user of the tool?

Reply:
Removed